### PR TITLE
Feature/te 416 drag and drop support for test step selector

### PR DESCRIPTION
**Pull request by georgportwich**

### CODE REVIEW

File: tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/editor/DropTargetListenerUpdateModelTest.xtend [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+import org.testeditor.aml.ComponentElement
+import org.testeditor.dsl.common.testing.AbstractTest
+import org.testeditor.dsl.common.testing.DslParseHelper
+import org.testeditor.tcl.ComponentTestStepContext
+import org.testeditor.tcl.TclModel
+import org.testeditor.tcl.dsl.TclStandaloneSetup
+import org.testeditor.tcl.dsl.ui.editor.DropTargetXtextEditorListener
+import com.google.common.io.ByteStreams
+
+class DropTargetListenerUpdateModelTest extends AbstractTest {
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/editor/DropTargetListenerUpdateModelTest.xtend]: This test class does not use the given/when/then pattern on which we agreed upon. Also it contains a lot of helper code which will make it very hard to maintain. This should be streamlined with our existing code base.

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		if (dndService == null)
+			return
+
+		val StyledText st = viewer.getTextWidget()
+
+		// Install drag target
+		dropTargetListener.editor = this
+		val DropTargetListener dropTargetListener = dropTargetListener
+		
+		dndService.addMergedDropTarget(st, DND.DROP_MOVE.bitwiseOr(DND.DROP_COPY), #[TextTransfer.instance],
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend]: Why do we need to overwrite `installTextDragAndDrop` in order to register a listener?

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.eclipse.ui.dnd.IDragAndDropService
+import javax.inject.Inject
+import org.testeditor.tcl.impl.TclFactoryImpl
+import org.testeditor.tsl.impl.TslFactoryImpl
+
+class DropTargetXtextEditor extends XtextEditor {
+
+	var ISourceViewer viewer
+	@Inject protected TclFactoryImpl tclFactory
+	@Inject protected TslFactoryImpl tslFactory
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend]: Those variables look unused to me.

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		val StyledText st = viewer.getTextWidget()
+
+		// Install drag target
+		dropTargetListener.editor = this
+		val DropTargetListener dropTargetListener = dropTargetListener
+		
+		dndService.addMergedDropTarget(st, DND.DROP_MOVE.bitwiseOr(DND.DROP_COPY), #[TextTransfer.instance],
+			dropTargetListener);
+	}
+	public def ISourceViewer getSourceViewerPublic() {
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend]: It is possible to increase the visibility of a method without giving it another name.

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+	@Inject protected TslFactoryImpl tslFactory
+
+	@Inject extension ILocationInFileProvider
+	@Inject protected ContentAssistContext.Factory contentAssistFactory
+	
+	@Accessors(PUBLIC_SETTER)
+	private DropTargetXtextEditor editor
+
+	override void dragEnter(DropTargetEvent event) {
+		if ("org.testeditor.tcl.dsl.Tcl" != editor.languageName) {
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: `org.testeditor.tcl.dsl.Tcl` can be retrieved using:
```
	@Inject @Named(Constants.LANGUAGE_NAME)
	String languageName
```

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+import org.testeditor.tsl.impl.StepContentTextImpl
+import org.testeditor.tcl.impl.SpecificationStepImplementationImpl
+import org.eclipse.xtext.ui.editor.contentassist.ContentAssistContext
+import org.eclipse.xtext.util.StringInputStream
+import java.util.Collections
+import org.eclipse.emf.ecore.resource.ResourceSet
+import org.eclipse.xtext.resource.XtextResource
+import java.net.URI
+
+class DropTargetXtextEditorListener extends DropTargetAdapter {
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: All methods in this class are public. Is this necessary? Also, public methods should have an explicit return type.

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+			event.detail = DND.DROP_NONE
+		}
+	}
+
+	override void drop(DropTargetEvent event) {
+
+		val List<String> toFormat = newArrayList
+		val List<String> currentElement = newArrayList
+
+		editor.document.modify(
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: If you write `editor.document.modify[updateModel(...)]` there is no need for the explicit creation of the anonymous type within `updateModel`

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+import org.eclipse.xtext.util.StringInputStream
+import java.util.Collections
+import org.eclipse.emf.ecore.resource.ResourceSet
+import org.eclipse.xtext.resource.XtextResource
+import java.net.URI
+
+class DropTargetXtextEditorListener extends DropTargetAdapter {
+
+	@Inject protected TclFactoryImpl tclFactory
+	@Inject protected TslFactoryImpl tslFactory
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: EMF factories should not be injected (or at least not their Impl class). Rather use `static val tclFactory = TclFactory.eINSTANCE`

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+			setCursorToNewElement(currentElement)
+		)
+
+	}
+	def updateModel(List<String> toFormat, List<String> currentElement) {
+
+		new IUnitOfWork.Void<XtextResource>() {
+
+			override process(XtextResource resource) throws Exception {
+				if (resource.contents.head instanceof TclModel) {
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: If you assign `resource.contents.head` to a variable beforehand, there is no need for the cast below. Xtend will automatically cast it for you within the `if` clause and the code will get more readable.

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+					val List<EObject> toFormatEObject = newArrayList
+					val TclModel tclModel = resource.contents.head as TclModel
+					var EObject dropTarget = findDropTarget(resource)
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: No need to repeat the types and `dropTarget` should be final (`val`).

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend (hunk 6 above)
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: Also the code is pretty hard to read for me. Could you restructure and reformat it a little?

File: tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/formatting2/TclFormatter.xtend [hunk 11]
```diff
@@ -127,9 +128,12 @@ class TclFormatter extends XbaseFormatter {
 		macroTestStepContext.steps.forEach[format]
 	}
 
-	def dispatch void format(TestStep testStep, extension IFormattableDocument document) {
+	def dispatch void format(AbstractTestStep testStep, extension IFormattableDocument document) {
 		testStep.regionFor.keyword("-").prepend[newLine]
-		testStep.contents.forEach[format]
+		if (testStep instanceof TestStep) {
+			testStep.contents.forEach[format]
+			testStep.regionFor.keyword(".").prepend[noSpace]
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/formatting2/TclFormatter.xtend]: +1 for addressing this issue, but could you do this in a different pull request? Also for each formatter test an accompanying test should be written.

File: rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelectorDragSourceListener.xtend [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.e4.core.di.annotations.Creatable
+import org.eclipse.jface.util.LocalSelectionTransfer
+
+@Creatable
+class TestStepSelectorDragSourceListener implements DragSourceListener {
+
+	@Inject AmlDropSupport amlDropSupport
+	@Inject TestStepSelectorDropTextProvider dropTextProvider
+	static val logger = LoggerFactory.getLogger(TestStepSelectorDragSourceListener);
```
Comment (georgportwich) [inline rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelectorDragSourceListener.xtend]: done

File: rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelectorDragSourceListener.xtend [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	override dragFinished(DragSourceEvent event) {
+		logger.info("drag stop")
+	}
+
+	override dragSetData(DragSourceEvent event) {
+		LocalSelectionTransfer.transfer.selection = viewer.structuredSelection
+		logger.info("set selection " + viewer.structuredSelection)
+		event.data = "test"
```
Comment (georgportwich) [inline rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelectorDragSourceListener.xtend]: Forgetten debug-code. Removed

File: rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelectorDragSourceListener.xtend [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+	@Inject TestStepSelectorDropTextProvider dropTextProvider
+	static val logger = LoggerFactory.getLogger(TestStepSelectorDragSourceListener);
+	TreeViewer viewer
+
+	public def void setViewer(TreeViewer viewer) {
+		this.viewer = viewer
+	}
+
+	override dragFinished(DragSourceEvent event) {
+		logger.info("drag stop")
```
Comment (georgportwich) [inline rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelectorDragSourceListener.xtend]: Not needed any more. Complete removed

File: rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelectorDragSourceListener.xtend (hunk 13 above)
Comment (georgportwich) [inline rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelectorDragSourceListener.xtend]: Class was complete refactored

File: rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelectorDragSourceListener.xtend [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	override dragSetData(DragSourceEvent event) {
+		LocalSelectionTransfer.transfer.selection = viewer.structuredSelection
+		logger.info("set selection " + viewer.structuredSelection)
+		event.data = "test"
+			if (TextTransfer.instance.isSupportedType(event.dataType) &&
```
Comment (georgportwich) [inline rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelectorDragSourceListener.xtend]: Class was complete refactored

File: tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/editor/DropTargetListenerUpdateModelTest.xtend (hunk 1 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/editor/DropTargetListenerUpdateModelTest.xtend]: Sorry I don't get it

File: tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/editor/DropTargetListenerUpdateModelTest.xtend [hunk 16]
```diff
@@ -0,0 +1,441 @@
...
+	@Before
+	def void parseAmlModel() {
+		var byte[] encoded = ByteStreams.toByteArray(this.getClass().getClassLoader().getResourceAsStream("test.aml"))
+		amlModel = parserHelper.parseAml(new String(encoded));
+
+		val tclInjector = (new TclStandaloneSetup).createInjectorAndDoEMFRegistration
+		serializer = tclInjector.getInstance(ISerializer)
+	}
+
+	@Inject protected extension DslParseHelper parserHelper
```
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/editor/DropTargetListenerUpdateModelTest.xtend]: done

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend (hunk 3 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend]: done

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend (hunk 2 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend]: How would you do it?

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend (hunk 4 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend]: No

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend (hunk 8 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: Why?

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend (hunk 5 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: How do I understand from the code, that the language name is tcl and not aml?

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend (hunk 7 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: Sorry but I don't know how to do it

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend (hunk 9 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: I tried it but it makes the code longer and less readable. I prefer an explicit type cast. So I have all Information in one line and I don’t have to check several lines to understand the code

File: tcl/org.testeditor.tcl.dsl.ui.tests/META-INF/MANIFEST.MF [hunk 17]
```diff
@@ -9,6 +9,9 @@ Require-Bundle: org.junit,
  org.eclipse.xtext.junit4,
  org.eclipse.xtext.xbase.junit,
  org.testeditor.tcl.dsl.tests,
- org.testeditor.dsl.common.testing
+ org.testeditor.dsl.common.testing,
+ org.eclipse.core.filesystem;bundle-version="1.6.0"
 Bundle-Vendor: Signal Iduna Corporation
+Import-Package: org.testeditor.aml.dsl.generator,
```
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui.tests/META-INF/MANIFEST.MF]: I tried it, but the bundle org.testeditor.aml.dsl ist no available

File: tcl/org.testeditor.tcl.dsl.ui.tests/resources/test.aml [hunk 18]
```diff
@@ -0,0 +1,70 @@
+package SwingDemo
```
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui.tests/resources/test.aml]: There is missing a lot. I would have to extend it to my needs. Do you think it makes sense?

File: tcl/org.testeditor.tcl.dsl.ui.tests/META-INF/MANIFEST.MF (hunk 17 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui.tests/META-INF/MANIFEST.MF]: done

File: tcl/org.testeditor.tcl.dsl.ui.tests/resources/test.aml (hunk 18 above)
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui.tests/resources/test.aml]: What kind of extension is necessary for testing drag&drop?

File: tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/editor/DropTargetListenerUpdateModelTest.xtend (hunk 1 above)
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/editor/DropTargetListenerUpdateModelTest.xtend]: There are two things I had in mind here:

1. Given / When / Then pattern - see most of the other tests plus http://martinfowler.com/bliki/GivenWhenThen.html
2. The helper methods might be useful elsewhere and/or can be simplified. Having those exclusively in this test will result in a test that is hard to maintain. Imagine if we change the internal model of AML / TCL and have this kind of code in every test-case. We would need to adapt each one of them.

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend (hunk 4 above)
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditor.xtend]: In this case "no" because it's a final method. I examined the code more closely and I think this method should be dropped completely. There exists already such a method called `getInternalSourceViewer` which is already used by `DropTargetXtextEditorListener`.

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend (hunk 8 above)
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: 1. `TclFactoryImpl` should be considered an "internal" class and not API, hence not used.
2. If you want to get the factory injected, you should inject `TclFactory` and create a proper Guice binding for it. Otherwise you'll create new factory instances all the time. You need to look at the source code closely if this is a problem or not (in this instance it should not be a problem).
3. More consistent with the rest of the code base

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend (hunk 5 above)
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: That's true but on the other hand: what if the name of the language is changed? The person doing that will probably not think of this part of the code. We could probably name the variable `tclLanguageName` to make it more clear.

Apart from that: if the listener is only registered on the Tcl editor, can this ever be not true?

File: rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelectorDragSourceListener.xtend [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+				logger.info("droped text: " + event.data)
+			} else {
+				event.data = ""
+			}
+	}
+
+	override dragStart(DragSourceEvent event) {
+		LocalSelectionTransfer.transfer.selection = viewer.structuredSelection
+		logger.info("set selection " + viewer.structuredSelection)
+		logger.info("drag start oho")
```
Comment (georgportwich) [inline rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelectorDragSourceListener.xtend]: Does not exist anymore

File: tcl/org.testeditor.tcl.dsl.ui.tests/resources/test.aml (hunk 18 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui.tests/resources/test.aml]: We need new interactions, new a Application

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend (hunk 9 above)
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: I disagree. Right now you have:
```
		if (resource.contents.head instanceof TclModel) {
			val TclModel tclModel = resource.contents.head as TclModel
```
(and the type cast is not even the first thing you do inside the if)

Instead if you use:
```
		val model = resource.contents.head
		if (model instanceof TclModel) {
```

You have actually less code and less repeated code. In the first example you repeat `resource.contents.head` twice.

File: tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/editor/DropTargetListenerUpdateModelTest.xtend (hunk 1 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui.tests/src/org/testeditor/tcl/dsl/ui/tests/editor/DropTargetListenerUpdateModelTest.xtend]: The given is about 25 line of code. The when is the inserted code. The when is 25 lines of code + the inserted code. This hardly fits on a screen and is difficult to read because you have to compare the given and the then line by line to understand the different. This was the result of the review with Gunther. To make the test readable I created an easy to read format where I combined the given and the when. 
To extract the methods to a helper class is a good idea but I did not find an appropriate class to place my code.

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend (hunk 6 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: I reduced the visibility of the methods. Due to the package path of the testclasses org.testeditor.tcl.dsl.ui.tests.editor it is not possible to set the tested methods to protected. Is there a reason for the “tests” in the path? 
We already spend a lot of time to restructure the code. To do further restructuring I need concrete ideas how to do it.

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend (hunk 10 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: done

File: tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/formatting2/TclFormatter.xtend (hunk 11 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/formatting2/TclFormatter.xtend]: done

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend (hunk 8 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: OK I got it. done

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend (hunk 5 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend]: The listener is registered to the DropTargetXtextEditor. This editor is currently only used for the tcl, but it could be used for the aml or the tsl editor as well. Do you plans to rename the tcl?
